### PR TITLE
Update pallas doc with paged_attention

### PR DESCRIPTION
**Pull request by wonjoolee95**
- Add paged_attention example and integration example

Preview changes at https://github.com/pytorch/xla/blob/bc70b366d5a5c9fc06cb5e41d782329c3170174f/docs/pallas.md

### CODE REVIEW

File: docs/pallas.md [hunk 1]
```diff
@@ -36,19 +36,42 @@ For simple kernels, the adoption is just as simple as one liner. For more compli
...
+#### Example usage
 ```python3
 # Use built-in kernels
 from torch_xla.experimental.custom_kernel import flash_attention
 output = flash_attention(q, k, v)
 ```
 
-You can just use it like any other torch.ops.
+#### Integration Example
+We have a fork of HuggingFace Llama 3 to demonstrate [a potential integration for FlashAttention here](https://github.com/pytorch-tpu/transformers/tree/alanwaketan/flash_attention).
```
Comment (JackCaoG) [inline docs/pallas.md:53]: https://github.com/pytorch/xla/blob/master/examples/flash_attention/train_decoder_only_flash_attention.py is a smaller example

File: docs/pallas.md [hunk 2]
```diff
@@ -36,19 +36,42 @@ For simple kernels, the adoption is just as simple as one liner. For more compli
...
+
+### PagedAttention
+
+#### Example usage
+```python3
+# Use built-in kernels
+from torch_xla.experimental.custom_kernel import paged_attention
```
Comment (JackCaoG) [inline docs/pallas.md:60]: can you just do `import torch_xla.experimental.custom_kernel`, I will soon move the paged attention to https://github.com/pytorch/xla/blob/master/torch_xla/core/custom_kernel.py which will be auto-registered

File: docs/pallas.md (hunk 1 above)
Comment (wonjoolee95) [reply docs/pallas.md]: Updated

File: docs/pallas.md (hunk 2 above)
Comment (wonjoolee95) [reply docs/pallas.md]: Updated

**Review by JackCaoG — APPROVED** [review]

**Pull request merged**